File: src/python/PythonSDK/foundationallm/hubs/agent/agent_hub_request.py
from pydantic import BaseModel
from typing import Optional
from foundationallm.models.orchestration import MessageHistoryItem

class AgentHubRequest(BaseModel):
    """     
    AgentHubRequest contains the information needed to retrieve AgentMetadata from the AgentHub.
    
    user_prompt: the prompt the user entered into the system.
    message_history: the current conversation history with the user.
        
    """
    user_prompt:str
    message_history: Optional[MessageHistoryItem] = None

File: src/dotnet/AgentFactory/Models/Messages/AgentHubRequest.cs
using FoundationaLLM.Common.Models.Chat;
using FoundationaLLM.Common.Models.Orchestration;
using Newtonsoft.Json;

namespace FoundationaLLM.AgentFactory.Core.Models.Messages
{
    /// <summary>
    /// Represents a request that is sent to an AgentHub to get lists of agents.
    /// </summary>
    public class AgentHubRequest : OrchestrationRequest
    {
        /// <summary>
        /// Conversation history of the current chat session.
        /// </summary>
        [JsonProperty("message_history")]
        public List<MessageHistoryItem>? MessageHistory { get; set; }
    }
}
